policy_agent/policy_agent.py

```diff
@@ -0,0 +1,643 @@
+
+        elif kind == "Pod":
+            resource_pool = self.pods
+            obj = Pod(target)
+
+        elif kind == "Endpoints":
+            resource_pool = self.endpoints
+            obj = Endpoints(target)
+
+        target_key = obj.key
+
+        if target_key in resource_pool:
+            del resource_pool[target_key]
+            _log.info("%s %s deleted from Calico store" % (kind, target_key))
+
+        else:
+            _log.error(
+                "Tried to Delete %s %s but it was not in bin" %
+                (kind, target_key))
+
+    def match_pod(self, namespace, name):
+        try:
+            return self.pods["%s/%s" % (namespace, name)]
+        except KeyError:
+            _log.error("Pod %s in NS %s not processed" % (name, namespace))
+            return None
+
+    def resync(self):
+        """
+        Updates Calico store with new resource information
+        """
+        # Create/update Namespace Profiles
+        for namespace_key, ns in self.changed_namespaces.items():
+            if ns.create_ns_profile():
+                self.nsRV = ns.RV
+                self.namespaces[namespace_key] = ns
+                del self.changed_namespaces[namespace_key]
+
+        # Process Services
+        for service_key, svc in self.changed_services.items():
+            self.svcRV = svc.RV
+            self.services[service_key] = svc
+            del self.changed_services[service_key]
+
+        # Apply Namespace policy to new/updated pods
+        for pod_key, po in self.changed_pods.items():
+            if po.apply_ns_policy():
+                if po.remove_default_profile():
+                    self.poRV = po.RV
+                    self.pods[pod_key] = po
+                    del self.changed_pods[pod_key]
+
+        # As endpoint lists change, create/add new profiles to pods as
+        # necessary
+        # TODO: if some object not ready for resync, finish loop w/out moving to
+        # processed list
+        for ep_key, ep in self.changed_endpoints.items():
+
+            # Get policy of the Service and namespace. Uses same namespace/name
+            # dict key.
+            try:
+                svc_policy = self.services[ep_key].type
+            except KeyError:
+                _log.warning("Service %s not yet in store" % ep_key)
+                continue
+
+            try:
+                ns_policy = self.namespaces[ep.namespace].policy
+            except KeyError, AttributeError:
+                _log.warning("Namespace %s not yet in store" % ep.namespace)
+                continue
+
+            _log.info("Using svc policy %s and ns policy %s" %
+                      (svc_policy, ns_policy))
+
+            # if namespace is open, or svc is closed, do nothing
+            if ns_policy == "Open" or svc_policy == "NamespaceIP":
+                self.endpoints[ep_key] = ep
+                del self.changed_endpoints[ep_key]
+
+            # else define service policy
```

Comment:
Don't stick newlines or comments between your if / else blocks.  In general commenting should look something like this:

```
# Make action based on whether or not certain condition is true because A, B, C. (This one is optional)
if condition:
  # The condition is true, we must do X because blah.
  do_x()
else:
  # The condition is not true, we must do Y because blah.
  do_y()
```